### PR TITLE
Use less deprecated methods in base.php

### PR DESCRIPTION
**Pull request by artonge**
Fix static analysis introduced by https://github.com/nextcloud/server/pull/30733

@nickvergessen can you confirm that this is working, I have no idea how to test the updater web UI :/

### CODE REVIEW

**Comment by nickvergessen** [comment]
> can you confirm that this is working, I have no idea how to test the updater web UI :/

The problem for me was quite easy:
* Change version.php
* Open web page
* Open console
* Try to press the update button

**Comment by artonge** [comment]
Tested and it work, but I had to extract some `$` call outside of a listener for `DOMContentLoaded` as it wasn't called. This should be safe as `jQuery` is set by `core-common.js`.

**Comment by nickvergessen** [comment]
> but I had to extract some $ call outside of a listener for DOMContentLoaded as it wasn't called for some reason. 

Works fine here even when I revert that change.

File: core/js/update.js [hunk 1]
```diff
@@ -146,22 +146,21 @@
 
 })();
 
-window.addEventListener('DOMContentLoaded', function() {
```
Comment (CarlSchwan) [inline core/js/update.js:149]: Why is this code in the typical `(function() {` from jquery?

Comment (nickvergessen) [reply core/js/update.js:149]: As per his comment:

> Tested and it work, but I had to extract some `$` call outside of a listener for `DOMContentLoaded` as it wasn't called for some reason. This should be safe as `jQuery` is set by `core-common.js`.

But yeah works locally with reverting this change

**Review by skjnldsv — APPROVED** [review]

File: core/js/update.js (hunk 1 above)
Comment (artonge) [reply core/js/update.js:149]: It works on chrome but not firefox :confused:. Can someone confirm the behaviour on firefox?

Comment (nickvergessen) [reply core/js/update.js:149]: It works for me in Chromium and Firefox

Comment (nickvergessen) [reply core/js/update.js:149]: So let's revert for now, so psalm is happy again and then investigate further what's wrong with your FF?

**Review by nickvergessen — APPROVED** [review]

**Review by CarlSchwan — APPROVED** [review]

**Comment by artonge** [comment]
CI failure is unrelated.

**Pull request merged**